app/views/how-tos/filters.html

```diff
@@ -0,0 +1,141 @@
+      that you can use or show the answers later. To change the format of how
+      these answers appear, you can apply filters. You can:</p>
+
+    <ul class="nshuk-list nhsuk-list-bullet">
+      <li>use existing Nunjucks filters</li>
+      <li>create a Nunjucks filter</li>
+      <li>use filters from other projects</li>
+    </ul>
+    <h2 class="nhsuk-heading-m">Use existing Nunjucks filters</h2>
+    <p>
+      You can
+      <a
+        href="https://mozilla.github.io/nunjucks/templating.html#builtin-filters"
+        >use existing Nunjucks filters</a
+      >
+      in the kit. For example, you can use the Nunjucks
+      <code>upper</code> filter to change the format of text to upper case:
+    </p>
+
+    <pre
+      class="app-pre"
+    ><code class="app-code">&#123;{ data['name'] | upper }}</code></pre>
+
+    <h2 class="nhsuk-heading-m">Create a Nunjucks filter</h2>
+    <p>
+      Add your own filters to the <code>filters.js</code> file. Filters are
+      written in JavaScript.
+    </p>
+    <pre
+      class="app-pre"
+    ><code class="app-code">module.exports = function (env) { /* eslint-disable-line func-names,no-unused-vars */
+  const filters = {};
+
+  filters.uppercase = function(content) {
+    return content.toUpperCase()
+  }
+  return filters;
+};</code>
+</pre>
+    <p>Then use it on a page like this:</p>
+    <pre
+      class="app-pre"
+    ><code class="app-code">&#123;{ data['name'] | uppercase }&#125;</code></pre>
+
+    <h3 class="nhsuk-heading-s">Use HTML in a Nunjucks filter</h3>
+    <p>If you want to use HTML in a filter, write the filter:</p>
+    <pre
+      class="app-pre"
+    ><code class="app-code">{{ " filters.bold = function (content) {
+  return '<strong>' + content + '</strong>';
+  // remember to use 'safe' after to get the HTML formatting
+}" | escape }} </code></pre>
+    <p>Then use it on a page with the extra filter <code>safe</code>:</p>
+    <pre
+      class="app-pre"
+    ><code class="app-code">{{"{{ data['name'] | bold | safe }}" | escape }}</code></pre>
+    <p>
+      Safe makes the filter format the HTML and not just show the code
+      <code>{{"<strong></strong>" | escape }}</code>.
+    </p>
+
+    <p>You can also use these filters together:</p>
+    <pre
+      class="app-pre"
+    ><code class="app-code">{{"{{ data['name'] | upper |  bold | safe }}" | escape }}</code></pre>
+
+    <h2 class="nhsuk-heading-m">Use filters from other projects</h2>
+    <p>
+      You can import filters from other projects like the <a href="https://x-govuk.github.io/govuk-prototype-filters/get-started/">x-govuk Prototype Filters
+      project</a>. If you want to do this, you will need to install it first - follow the instructions for 'Using with earlier versions of the Prototype Kit'.
+    </p>
+
+    <p>For example, you can <a href="https://x-govuk.github.io/govuk-prototype-filters/get-started/">use existing x-govuk date filters</a> to add dates that will change in your prototype. </p>
```

Comment:
I had half a mind to add a date filter as a built-in filter to the NHS prototype kit, would make this a bit redundant (although the other x-govuk filters can still be helpful).

Using the `govukDate` filter is a bit fiddly as the instructions suggest you should use another filter to parse the date from the date input before handing it over to the `govukDate` filter, like this:

```njk
{{ data | isoDateFromDateInput("dob") | govukDate }}
```

However since [a recent change](https://github.com/nhsuk/nhsuk-frontend/pull/994) I made to the default date input names in the NHS Frontend, it's a bit simpler in the NHS Prototype Kit, as you’d only have to do this:

```njk
{{ data.dob | isoDateFromDateInput | govukDate }}
```

but I think we could simplify even more and skip the middle filter by allowing the date formatter to accept an object or an ISO string, which would you let you do this:

```njk
{{ data.dob | nhsDate }}
```

Reply:
ooh that looks good! As you say, I do agree in that I wonder if it's better to downplay the xgov filters for now and try and build something for ours (though that also does make me think about whether it's worth upgrading the filters in general).

Will have a play on this - arguably I could just remove all the date stuff for now apart from saying it's available (which was the original example in the govuk prototype kit) and make into a separate ticket (even if just added to this page).

Reply:
Yeah that sounds good - the x-govuk ones have enough gotchas that it's probably better to downplay for now?

Reply:
The filters I did on register [did similar optimisations](https://github.com/DFE-Digital/register-trainee-teachers-prototype/blob/2c058586d09d9abebd6349a67f4d28fc53dd3fac/app/filters/dates.js) so you don't need to chain them.

Reply:
Note those are using Moment, which is probably not good to adopt any more. For newer stuff, I'm now using [Day.js](https://day.js.org/)

Reply:
Notes about dates functionality made into a separate ticket https://github.com/nhsuk/nhsuk-prototype-kit/issues/433 - I'm leaving this as prototype kit only as there is no point in adding documentation until we have a direction.

Reply:
Have also simplified the page

Reply:
also just to clarify @edwardhorsford I think your comment was about not chaining govuk date filters, rather than that I can use the filter to default have html formatting (which I don't think I can without us changing the filters functionality to be aligned with the current govuk prototype filters)

Reply:
I was kinda hoping that nowadays you could use the [Intl.DateTimeFormat api built in to JavaScript and Node](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) without requiring a third party library?

Reply:
@frankieroberto is this something that needs to stay in this ticket or can be part of #433 ? Even if we add it (which I think we can, as you say, between Ed's stuff and even stuff I put together in other repos) it's still going to need to be tested and ideally just put in the kit even jsut in filters.js

Reply:
Spun out a separate ticket sounds good to me.

Reply:
Cool - will resolve this with separate issue for date in https://github.com/nhsuk/nhsuk-prototype-kit/issues/433